src/services/filesync/filesync.ts

```diff
@@ -573,6 +590,31 @@ export class FileSync {
     ]);
   }
 
+  private async _getHashes(): Promise<{
+    gadgetFilesVersion: bigint;
+    filesVersionHashes: Hashes;
+    localHashes: Hashes;
+    gadgetHashes: Hashes;
+  }> {
+    const [localHashes, filesVersionHashes, { gadgetFilesVersion, gadgetHashes }] = await Promise.all([
+      // get the hashes of our local files
+      this.directory.hashes(),
+      // get the hashes of the files at our current filesVersion
+      this.filesVersion === 0n
+        ? {}
+        : this.editGraphQL
+            .query({ query: FILE_SYNC_HASHES_QUERY, variables: { filesVersion: String(this.filesVersion) } })
+            .then((data) => data.fileSyncHashes.hashes),
+      // get the hashes of the files at the latest filesVersion
+      this.editGraphQL.query({ query: FILE_SYNC_HASHES_QUERY }).then((data) => ({
+        gadgetFilesVersion: BigInt(data.fileSyncHashes.filesVersion),
+        gadgetHashes: data.fileSyncHashes.hashes,
+      })),
```

Comment:
If your local files version matches the files version in Gadget, do we generate the hashes twice?

Reply:
If this was just one graphQL query where we passed along the `localFilesVersion` we could skip generating 50% of the hashes a lot of the time

Reply:
Right, yeah we can turn this into a single query 👍

Reply:
Did this here: [Use withoutUnnecessaryChanges inside getChanges](https://github.com/gadget-inc/ggt/pull/1027/commits/81d0156b338291f9f1f02d6f2caec5a0de062133)

Reply:
Did this here: [Use fileSyncComparisonHashes](https://github.com/gadget-inc/ggt/pull/1027/commits/69dedaf56b07d83a461cd62b3399469b85a79698)